File: src/javascript/TwoSum.js
var twoSum = function(nums, target) {
    const values = new Map()

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i]
        const matchingValue = values.get(complement)
        if (matchingValue !== undefined) {
            return [matchingValue, i]
        }
        values.set(nums[i], i)
    }
}

const arr = [2, 7, 11, 15];
console.log(twoSum(arr, 9));

File: src/TwoSum.java
import java.util.*;

public class TwoSum {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> complement = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (complement.keySet().contains(nums[i])) {
                return new int[]{complement.get(nums[i]), i};
            }
            complement.put(target - nums[i], i);
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        int[] nums = new int[]{2, 7, 11, 15};
        int target = 9;
        int[] indices = twoSum(nums, target);
        System.out.println(Arrays.toString(indices));
    }
}
